Solution

pashawarganov opened the pull request:
write 3 functions

pashawarganov committed:
Solution
write 3 functions

pashawarganov committed:
Modified style
flake8 errors:
E124
E501

MNDonut commented on app/main.py:
```diff
@@ -1,13 +1,21 @@
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {"errors": [format_linter_error(error) for error in errors],
+            "path": file_path,
+            "status": "failed" if errors else "passed"
```
move everything besides the first bracket on the next line

MNDonut commented on app/main.py:
```diff
@@ -1,13 +1,21 @@
+    return {"errors": [format_linter_error(error) for error in errors],
+            "path": file_path,
+            "status": "failed" if errors else "passed"
+            }
```
1 tabulation is redundant. follow the first example from checklist

MNDonut commented on app/main.py:
```diff
@@ -1,13 +1,21 @@
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [format_single_linter_file(key, value)
+            for key, value in linter_report.items()
+            ]
```
the same as above

pashawarganov committed:
Second modified style
- move everything besides the first bracket on the next line
- delete 1 tabulation

LiudmylaKulinchenko commented on app/main.py:
```diff
@@ -1,13 +1,23 @@
-    pass
+    return [
+        format_single_linter_file(key, value)
+        for key, value in linter_report.items()
```
Change key/value variable names here, please, to make them more descriptive

pashawarganov committed:
rename 'key, value' to 'path, errors'
